### PR TITLE
examples/shader: Add the chromatic aberration example

### PR DESCRIPTION
**Pull request by tomlister**
#1284

### CODE REVIEW

**Comment by tomlister** [comment]
Sorry about all the commits. Have had a long day.

File: examples/shader/chromaticaberration.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+// +build ignore
+
+package main
+
+var Time float
+var Cursor vec2
+
+func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
+	center := vec2(textureDstSize().x/4, textureDstSize().y/4)
```
Comment (hajimehoshi) [inline examples/shader/chromaticaberration.go:23]: This is not a center unfortunately. A texture size can be much bigger than the exposed image. To get a precise center, you'd need to pass the size as a uniform variable.

File: examples/shader/chromaticaberration.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	amount := normalize(center-Cursor).x / 100
+	clr := vec3(0, 0, 0)
```
Comment (hajimehoshi) [inline examples/shader/chromaticaberration.go:25]: `var clr vec3`

File: examples/shader/chromaticaberration.go (hunk 1 above)
Comment (hajimehoshi) [reply examples/shader/chromaticaberration.go:23]: So I'd recommend to add `var Size vec2` to every shader in this example and pass the image size.

File: examples/shader/chromaticaberration.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// +build ignore
+
+package main
+
+var Time float
+var Cursor vec2
+var ScreenSize vec2
+
+func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
+	center := vec2(ScreenSize.x/2, ScreenSize.y/2)
```
Comment (hajimehoshi) [inline examples/shader/chromaticaberration.go:24]: `center := ScreenSize / 2`

File: examples/shader/main.go [hunk 4]
```diff
@@ -121,6 +122,7 @@ func (g *Game) Draw(screen *ebiten.Image) {
 	op.Uniforms = []interface{}{
 		float32(g.time) / 60,                // Time
 		[]float32{float32(cx), float32(cy)}, // Cursor
+		[]float32{float32(w), float32(h)},   //Width, Height
```
Comment (hajimehoshi) [inline examples/shader/main.go:125]: `// ScreenSize`

File: examples/shader/chromaticaberration.go (hunk 3 above)
Comment (tomlister) [reply examples/shader/chromaticaberration.go:24]: Sorry still getting used to shader coding. I'll get it fixed now

**Comment by tomlister** [comment]
I'll be writing some more tonight and tommorow!

**Pull request merged**